src/main/scala/com/yotpo/metorikku/output/writers/file/FileOutputWriter.scala

```diff
@@ -104,6 +107,13 @@ class FileOutputWriter(props: Map[String, Object], outputFile: Option[File]) ext
 
         writer.save()
 
+        fileOutputProperties.protectFromEmptyOutput match {
+          case Some(true) => {
+            if (catalog.tableExists(tableName) && dataFrame.sparkSession.table(tableName).head(1).isEmpty) {
```

Comment:
I don't think that's good enough... You need to read the output folder...
something like spark.read.format(...)....